### PR TITLE
Added web3.py

### PR DESCRIPTION
**Pull request by step21**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/web3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+build:
+  number: 0
+  skip: true  # [win]
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (hmaarrfk) [inline recipes/web3/meta.yaml:20]:
```suggestion
    - python <3.8
```

File: recipes/web3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: f378a773886122d2126d7ba81e9f5b6d2e7b4def7e88543699bf85a951835942
+
+build:
+  number: 0
+  skip: true  # [win]
```
Comment (hmaarrfk) [inline recipes/web3/meta.yaml:14]:
```suggestion
  skip: true  # [win or py==38]
```

File: recipes/web3/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - typing-extensions >=3.7.4.1,<4  # [py<38]
+    - websockets >=8.1.0,<9.0.0
+
+test:
+  imports:
+    - web3
+    - ens
+    - ethpm
+  commands:
+    - pip check
```
Comment (hmaarrfk) [inline recipes/web3/meta.yaml:45]: This is not a common check of correctness in conda. 

Can you explain what you are attempted to verify here?

File: recipes/web3/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  skip: true  # [win]
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools-markdown
+  run:
+    - python <3.8
```
Comment (hmaarrfk) [inline recipes/web3/meta.yaml:23]:
```suggestion
    - python
```

File: recipes/web3/meta.yaml (hunk 3 above)
Comment (step21) [reply recipes/web3/meta.yaml:45]: grayskull did this. I will remove it and maybe add the proper tests.

File: recipes/web3/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - eth-account >=0.5.2,<0.6.0
+    - eth-hash >=0.2.0,<1.0.0
+    - eth-typing >=2.0.0,<3.0.0
+    - eth-utils >=1.8.4,<2.0.0
+    - hexbytes >=0.1.0,<1.0.0
+    - ipfshttpclient >=0.4.13,<1
+    - jsonschema >=3.2.0,<4.0.0
+    - lru-dict >=1.1.6,<2.0.0
+    - protobuf >=3.10.0,<4
+    # - pypiwin32 >=223  # [win]
```
Comment (hmaarrfk) [inline recipes/web3/meta.yaml:34]: do you want to try windows builds now?

Comment (step21) [reply recipes/web3/meta.yaml:34]: no. they need pypiwin32 which is just a link to pywin32 but not on conda. In order not to patch upstream requirements, I asked in the pywin32 feedstock if they could make an empty link package as a separate output, but no reply yet.

Comment (hmaarrfk) [reply recipes/web3/meta.yaml:34]: Is this just something you saw in their setup.py file? If so, those requirements are effectively ignored during install.

Lets try building and see what happens?

You can depend on `pywin32` and see what happens.

Add something to this effect as a comment, and you can cross reference your upstream issue as well.

Comment (step21) [reply recipes/web3/meta.yaml]: yes me and grayskull saw it. But also check pypiwin32 myself. (I mean the package, not what the code does as my windows access is limited)

File: recipes/web3/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: f378a773886122d2126d7ba81e9f5b6d2e7b4def7e88543699bf85a951835942
+
+build:
+  number: 0
+  skip: true  # [py==38]
```
Comment (hmaarrfk) [inline recipes/web3/meta.yaml:14]: Maybe just add a comment to yourself as to why you are skipping python 3.8 referencing some issue you raised on github.

**Comment by hmaarrfk** [comment]
maybe just try unpinning hypothesis in eth-abi and seeing what happens.

**Comment by hmaarrfk** [comment]
It seems to be related to test time, not runtime.

`eth` stuff seems to pin pretty hard to major versions. Not fun when you are downstream trying to package things.

Anyway this seems good to go, and the rest you can fix in the feedstock.

**Comment by step21** [comment]
yeah might just try it. It might pin pretty hard, as especially for defining abi and other primitives, if other packages update and behavior changes, it can have huge consequences. Plus I think the python team is relatively small afaik.

**Comment by step21** [comment]
> Anyway this seems good to go, and the rest you can fix in the feedstock.

ok

**Comment by hmaarrfk** [comment]
This is ready for review by somebody that can merge!

**Comment by step21** [comment]
@conda-forge/python please check/merge.

**Comment by isuruf** [comment]
the correct teams to ask for review in this repo are `conda-forge/help-python` or `conda-forge/staged-recipes`

**Comment by step21** [comment]
> the correct teams to ask for review in this repo are `conda-forge/help-python` or `conda-forge/staged-recipes`

ok thanks. Sorry!

**Comment by hmaarrfk** [comment]
Yeah, i guess i didn't want to ping people too much.

Maybe i should suggest to obviously linux users to build locally while somebody comes along and checks these.

It takes ~1 week to get somebody's attention. Thanks @isuruf

**Pull request merged**